### ISSUE TITLE
Throw error on invalid Swagger spec

### ISSUE THREAD

**Issue by petrenkotino** [Springworks/hapi-sov#5]
I had something similar to this in my swagger spec:
```
 responses: {
      200: {
        description: 'Vehicles',
        schema: {
          type: 'array',
          items: {
              schema: vehicle_entity_schema,
           },
        },
      },
}
```
Which is wrong because the `items` param doesn't want an object with a `schema` it just wants the schema object directly.
What happened is the tests passed as if there's no problem, but I caught the error otherwise. After fixing the error, a bunch of tests started failing because of responses not matching specs.

**Comment by nimerritt** [comment]
Yeah, we should run the Swagger validator over the schema before using it. There is a JSON schema for Swagger which covers most of the validation. There are additional restrictions which are not covered by the schema, but it is probably sufficient for what we're doing.

**Comment by nimerritt** [comment]
https://github.com/OAI/OpenAPI-Specification/blob/master/schemas/v2.0/schema.json

**Comment by nimerritt** [comment]
Maybe use https://github.com/BigstickCarpet/swagger-parser as it also resolves references, but it might be overkill.
